android/engine/src/main/java/org/smartregister/fhircore/engine/ui/theme/Themes.kt

```diff
@@ -16,16 +16,59 @@
 
 package org.smartregister.fhircore.engine.ui.theme
 
+import androidx.appcompat.app.AppCompatDelegate
 import androidx.compose.material.MaterialTheme
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
 
 /**
  * TODO fix issue with ktfmt formatting annotated high order functions. Current workaround below:
  * lambda in this format content: (@Composable() () -> Unit) to allow spotlessApply
  *
  * To enable app theme set darkTheme = isSystemInDarkTheme
  */
+//@Composable
+//fun AppTheme(darkTheme: Boolean = false, content: (@Composable() () -> Unit)) {
+//  MaterialTheme(colors = if (darkTheme) DarkColors else LightColors, content = content)
+//}
+
+class ThemeViewModel : ViewModel() {
+  var isSystemInDarkTheme by mutableStateOf(false)
+    private set
+
+  fun updateTheme(isDarkTheme: Boolean) {
+    isSystemInDarkTheme = isDarkTheme
+    val theme = if (isSystemInDarkTheme) AppCompatDelegate.MODE_NIGHT_YES else AppCompatDelegate.MODE_NIGHT_NO
+    AppCompatDelegate.setDefaultNightMode(theme)
```

Comment:
From the documentation `AppCompatDelegate.setDefaultNightMode()` defaults to `AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM` which I think should be the intended behavior even future when we decide to suppor the Dark theme meaning this would be unnecessary unless it's the workaround for fixing the app crash.